packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -650,6 +709,47 @@ export class RemotedFluidDataStoreContext extends FluidDataStoreContext {
 
         return this.details;
     }
+
+    /**
+     * This should be called when running garbage collection to get details needed for
+     * idetifying which fluid objects to collect.
+     */
+    protected async getGarbageCollectioDetails(): Promise<IGarbageCollectionDetails> {
```

Comment:
I'm not sure if design similar to getInitialSnapshotDetails() would work here. In getInitialSnapshotDetails case, the data does not change over time - it returns mostly pkg, but also initial snapshot (not latest snapshot!).
Whereas here, this function (if I got you right) will in future return GC info - current info (or rather, info based on latest state). Partitioning (having different impls) between remote & local contexes will not work, as local context can't keep returning initial state - all impls will need to return latest state.

Maybe I'm wrong, but it's hard to see the future (i.e. access this design without seeing more info in the same blob and how its used through time).

I feel like you prior commit was good for time being, and possibly latest commit is good next step when we start having more scaffolding around data reuse. Such step-by-step approach would not impact back compat really, so it's save to move in smaller steps.

Reply:
I added a new blob because we will need a new one anyway. But what you said makes sense, we might as well add it once the details around it are more fleshed out.